src/panels/config/integrations/integration-panels/zwave_js/zwave_js-logs.ts

```diff
@@ -0,0 +1,148 @@
+
+  private _textarea?: HTMLTextAreaElement;
+
+  public hassSubscribe(): Array<UnsubscribeFunc | Promise<UnsubscribeFunc>> {
+    return [
+      subscribeZWaveJSLogs(this.hass, this.configEntryId, (log) => {
+        if (this._textarea) {
+          if (Array.isArray(log.message)) {
+            log.message.map((line) => {
+              this._textarea!.value += `${line}\n`;
+              return null;
+            });
+          } else {
+            this._textarea!.value += `${log.message}\n`;
+          }
+        }
+      }),
+    ];
+  }
+
+  protected render() {
+    return html`
+      <hass-tabs-subpage
+        .hass=${this.hass}
+        .narrow=${this.narrow}
+        .route=${this.route}
+        .tabs=${configTabs}
+      >
+        <ha-card>
+          <div class="card-header">
+            <h1>
+              ${this.hass.localize("ui.panel.config.zwave_js.logs.title")}
+            </h1>
+          </div>
+          <div class="card-content">
+            ${this._logConfig
+              ? html`
+                  <paper-dropdown-menu
+                    dynamic-align
+                    .label=${this.hass.localize(
+                      "ui.panel.config.zwave_js.logs.log_level"
+                    )}
+                  >
+                    <paper-listbox
+                      slot="dropdown-content"
+                      .selected=${this._logConfig.level}
+                      attr-for-selected="value"
+                      @iron-select=${this._dropdownSelected}
+                    >
+                      <paper-item value="error">Error</paper-item>
+                      <paper-item value="warn">Warn</paper-item>
+                      <paper-item value="info">Info</paper-item>
+                      <paper-item value="verbose">Verbose</paper-item>
+                      <paper-item value="debug">Debug</paper-item>
+                      <paper-item value="silly">Silly</paper-item>
+                    </paper-listbox>
+                  </paper-dropdown-menu>
+                `
+              : ""}
+          </div>
+        </ha-card>
+        <textarea></textarea>
+      </hass-tabs-subpage>
+    `;
+  }
+
+  protected firstUpdated(changedProps) {
+    super.firstUpdated(changedProps);
+    this._textarea = this.shadowRoot!.querySelector("textarea")!;
+    this._fetchData();
+  }
+
+  private async _fetchData() {
+    if (!this.configEntryId) {
+      return;
+    }
+    this._logConfig = await fetchLogConfig(this.hass!, this.configEntryId);
+  }
+
+  private _dropdownSelected(ev) {
+    if (ev.target === undefined || this._logConfig === undefined) {
+      return;
+    }
+    if (this._logConfig.level === ev.target.selected) {
+      return;
+    }
+    setLogLevel(this.hass!, this.configEntryId, ev.target.selected);
+  }
+
+  static get styles(): CSSResultArray {
+    return [
+      haStyle,
+      css`
+        textarea {
+          width: 90%;
+          height: 90%;
```

Comment:
How will this look on mobile, should we have a minimum height?

Reply:
Also, what is 90%… are you assuming that the config card is 10% ?

Reply:
![image](https://user-images.githubusercontent.com/7545841/116311457-45458f00-a779-11eb-83ad-21ba80f19e7f.png)

Here's a screenshot from my phone

Reply:
just picked 90% to give it some spacing between the edges of the container

Reply:
that's for the width, but my comment is about the height. If the logs don't fit, it will now overflow? and then you scroll the whole page, and then the textarea?

Reply:
yeah there's a little bit of double scrolling on mobile

Reply:
Use flex box to make it fill the screen, so you don't have double scrolling. Check `unused entities` for example